### PR TITLE
Remove LegacyESVersion.V_6_5_x constants

### PR DESCRIPTION
**Pull request by adnapibar**
### Description
This PR removes LegacyESVersion.V_6_5_x constants.
 
### Issues Resolved
Part of the meta issue #1674
 
### Check List
- [x] All tests pass
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

Signed-off-by: Rabi Panda <adnapibar@gmail.com>

### CODE REVIEW

File: qa/full-cluster-restart/src/test/java/org/opensearch/upgrades/FullClusterRestartSettingsUpgradeIT.java [hunk 1]
```diff
@@ -1,126 +0,0 @@
...
-import static org.opensearch.transport.RemoteClusterService.SEARCH_REMOTE_CLUSTER_SKIP_UNAVAILABLE;
-import static org.opensearch.transport.SniffConnectionStrategy.SEARCH_REMOTE_CLUSTERS_SEEDS;
-import static org.opensearch.transport.SniffConnectionStrategy.SEARCH_REMOTE_CLUSTERS_PROXY;
-import static org.opensearch.transport.SniffConnectionStrategy.REMOTE_CLUSTERS_PROXY;
-import static org.hamcrest.Matchers.equalTo;
-
-public class FullClusterRestartSettingsUpgradeIT extends AbstractFullClusterRestartTestCase {
-
-    public void testRemoteClusterSettingsUpgraded() throws IOException {
-        assumeTrue("settings automatically upgraded since 6.5.0", getOldClusterVersion().before(LegacyESVersion.V_6_5_0));
```
Comment (VachaShah) [inline qa/full-cluster-restart/src/test/java/org/opensearch/upgrades/FullClusterRestartSettingsUpgradeIT.java:61]: Wouldn't we remove just this line and keep the remaining test intact?

File: server/src/internalClusterTest/java/org/opensearch/index/seqno/PeerRecoveryRetentionLeaseCreationIT.java [hunk 2]
```diff
@@ -78,7 +77,7 @@ public void testCanRecoverFromStoreWithoutPeerRecoveryRetentionLease() throws Ex
                         // simulate a version which supports soft deletes (v6.5.0-and-later) with which this node is compatible
                         VersionUtils.randomVersionBetween(
                             random(),
-                            Version.max(Version.CURRENT.minimumIndexCompatibilityVersion(), LegacyESVersion.V_6_5_0),
+                            Version.max(Version.CURRENT.minimumIndexCompatibilityVersion(), Version.CURRENT),
```
Comment (VachaShah) [inline server/src/internalClusterTest/java/org/opensearch/index/seqno/PeerRecoveryRetentionLeaseCreationIT.java:80]: Should it be `Version.CURRENT` or `LegacyESVersion.V_7_0_0`?

File: qa/full-cluster-restart/src/test/java/org/opensearch/upgrades/FullClusterRestartSettingsUpgradeIT.java (hunk 1 above)
Comment (adnapibar) [reply qa/full-cluster-restart/src/test/java/org/opensearch/upgrades/FullClusterRestartSettingsUpgradeIT.java:61]: Removing the assumption fails the test. So adding the assumption back. The entire test is to be removed as part of a new PR that removes the `search.remote.*` settings.

**Comment by adnapibar** [comment]
Transient tast failure - created issue #1806

**Comment by adnapibar** [comment]
start gradle check

**Review by andrross — APPROVED** [review]

**Review by reta — APPROVED** [review]

**Pull request merged**